src/group_theory/order_of_element.lean

```diff
@@ -210,6 +210,22 @@ lemma order_of_eq_prime {p : ℕ} [hp : fact p.prime]
   (hg : a^p = 1) (hg1 : a ≠ 1) : order_of a = p :=
 (hp.out.2 _ (order_of_dvd_of_pow_eq_one hg)).resolve_left (mt order_of_eq_one_iff.1 hg1)
 
+lemma order_of_eq_order_of_iff {β : Type*} [monoid β] {b : β} :
+  order_of a = order_of b ↔ ∀ n : ℕ, a ^ n = 1 ↔ b ^ n = 1 :=
+begin
+  simp_rw ← order_of_dvd_iff_pow_eq_one,
+  exact ⟨λ h n, by rw h, λ h, nat.dvd_antisymm ((h _).mpr (dvd_refl _)) ((h _).mp (dvd_refl _))⟩,
+end
+
+@[simp] lemma order_of_submonoid {G : Type*} [monoid G] {H : submonoid G} (σ : H) :
```

Comment:
Does this generalize to any elements related by an injective monoid_hom, in this case `submonoid.subtype`?

Reply:
Yes. I've added a new lemma.